solvers/lightning.py

```diff
@@ -0,0 +1,60 @@
+import warnings
+from benchopt import BaseSolver
+from benchopt import safe_import_context
+
+
+with safe_import_context() as import_ctx:
+    import numpy as np
+    from lightning.regression import SDCARegressor
+    from sklearn.exceptions import ConvergenceWarning
+
+
+class Solver(BaseSolver):
+    name = 'lightning'
+
+    install_cmd = 'conda'
+    requirements = ['sklearn-contrib-lightning']
+    references = [
+        'M. Blondel, K. Seki and K. Uehara, '
+        '"Block coordinate descent algorithms for large-scale sparse '
+        'multiclass classification" '
+        'Mach. Learn., vol. 93, no. 1, pp. 31-52 (2013)'
+    ]
+
+    def set_objective(self, X, y, l1_ratio, lmbda, fit_intercept):
+        self.X, self.y, self.l1_ratio, self.lmbda = X, y, l1_ratio, lmbda
+        self.fit_intercept = fit_intercept
+
+        self.clf = SDCARegressor(alpha=self.lmbda,
+                                 l1_ratio=self.l1_ratio, tol=0)
```

Comment:
It is better to use a very low tol like 1e-12, avoiding weird effects with the solver getting stuck for a very long time on subproblems